cloudsdk/src/main/java/io/particle/android/sdk/cloud/ParticleDevice.java

```diff
@@ -406,6 +406,31 @@ public void executeFlashingChange() throws RetrofitError {
         });
     }
 
+    @WorkerThread
+    public void flashCodeFile(final File file) throws ParticleCloudException {
+        performFlashingChange(new FlashingChange() {
+            @Override
+            public void executeFlashingChange() throws RetrofitError {
+                mainApi.flashFile(deviceState.deviceId,
+                        new TypedFile("multipart/form-data", file));
+            }
+        });
+    }
+
+
+    @WorkerThread
```

Comment:
OK, I get it now,

I have actually imitated the methods already there for `flashBinaryFile()`. You have a method for `File` and another one for `InputStream`. I am building an app where you write code for photon using Blockly, whose code output is a `String`. To sent the `String` code to photon, I converted the string into an `InputStream` instead of writing it to a `File`.